pxd_fastpath.c

```diff
@@ -1119,11 +1131,20 @@ void enableFastPath(struct pxd_device *pxd_dev, bool force)
 	memset(fp->device_path, 0, sizeof(fp->device_path));
 
 	pxd_dev->fp.fastpath = false;
-	pxd_resume_io(pxd_dev);
+	/// volume still remains suspended waiting for CLEANUP request to reopen IO.
```

Comment:
this is the case which needs a ioctl cleanup... because SAN path got established putting targets to handle IO in fastpath... but IO will immediately flow through native path without suspending and waiting for cleanup.

Reply:
So why can't pxd_fastpath_vol_cleanup() be called from here directly instead of relying on the ioctl?

Reply:
If the px volume setup this late fails to setup in fastpath, then immediately setting the volume in native path, can start receiving IOs but the target has been configured for exporting target dev/file and file handle is closed. So processing this IO at the target will crash since IOs are received for closed file handles. px does a SAN cleanup and then reopens the px volume to allow IOs.